test/sqllogictest/regex.slt

```diff
@@ -493,9 +493,8 @@ ORDER BY 1
 {hello,world}
 {hello,world}
 
-# TODO: Correct ordering when database-issues#4764 is fixed.
 query T
-SELECT foo FROM regexp_split_to_table('the quick brown fox jumps over the lazy dog', '\s+') AS foo ORDER BY 1
+SELECT foo FROM regexp_split_to_table('the quick brown fox jumps over the lazy dog', '\s+') WITH ORDINALITY AS t(foo, n) ORDER BY n
```

Comment:
@ggevay The order still doesn't look correct to me! Should we reopen the issue?

Reply:
This is due to this issue: https://github.com/MaterializeInc/materialize/blob/73075adf987b59f5668366360dd009b691ab6c7b/src/sql/src/func.rs#L1674

My plan is to disable `WITH ORDINALITY` for table functions in this category for now. I've added a Sentry error for when people use one of these functions with `WITH ORDINALITY`. If it will show no usage when it's rolled out this Thursday, I'll open a PR on Friday to disable `WITH ORDINALITY` for these table functions.

Reply:
Should we open a new issue to track that or is it never planned to be fixed?

Reply:
I'll open an issue for it. I would guess that it will come up sooner or later, but maybe not this year.